infrastructure/core/main.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/ec2"
+	"github.com/pulumi/pulumi-eks/sdk/v2/go/eks"
+	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes"
+	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
+	helmv3 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/helm/v3"
+	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+	"golang.org/x/exp/rand"
+)
+
+func generatePassword(length int) string {
+	source := rand.NewSource(uint64(time.Now().UnixNano()))
+	r := rand.New(source)
+
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789" +
+		"!@#$%^&*()_+")
+	password := make([]rune, length)
+	for i := range password {
+		password[i] = chars[r.Intn(len(chars))]
+	}
+	return string(password)
+}
+
+func main() {
+	namespaces := []string{"development", "paper", "live", "kubeflow", "monitoring"}
+
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		cfg := config.New(ctx, "")
+		minClusterSize, err := cfg.TryInt("minClusterSize")
+		if err != nil {
+			minClusterSize = 3
+		}
+		maxClusterSize, err := cfg.TryInt("maxClusterSize")
+		if err != nil {
+			maxClusterSize = 6
+		}
+		desiredClusterSize, err := cfg.TryInt("desiredClusterSize")
+		if err != nil {
+			desiredClusterSize = 3
+		}
+		eksNodeInstanceType, err := cfg.Try("eksNodeInstanceType")
+		if err != nil {
+			eksNodeInstanceType = "t3.small"
+		}
+		vpcNetworkCidr, err := cfg.Try("vpcNetworkCidr")
+		if err != nil {
+			vpcNetworkCidr = "10.0.0.0/16"
+		}
+
+		eksVpc, err := ec2.NewVpc(ctx, "eks-vpc", &ec2.VpcArgs{
```

Comment:
Nitpick: it's Go's idiom to fully capitalize acronyms in camelCASE - e.g. `eksVPC`.

Reply:
fixed